apps/src/dance/DanceVisualizationColumn.jsx

```diff
@@ -67,27 +74,38 @@ class DanceVisualizationColumn extends React.Component {
   };
 
   state = {
-    filterOff: this.setFilterStatus()
+    filterOn: this.getFilterStatus()
   };
 
   /*
     Turn the song filter off
   */
   turnFilterOff = () => {
-    this.setState({filterOff: true});
+    this.setState({filterOn: false});
   };
 
   /*
     The filter defaults to on. If the user is over 13 (identified via account or anon dialog), filter turns off.
    */
-  setFilterStatus() {
+  getFilterStatus() {
+    const {userType, under13} = this.props;
+
+    // Check if song filter override is triggered and initialize song filter to true.
+    const songFilter = songFilterOn();
+    if (songFilter) {
+      return true;
+    }
+
     // userType - 'teacher', 'student', 'unknown' - signed out users.
-    // under13 - boolean for signed in user representing age category. Teacher assumed > 13.
-    const signedInOver13 =
-      this.props.userType === 'teacher' ||
-      (this.props.userType === 'student' && !this.props.under13);
-    const signedOutAge = signedOutOver13();
-    return signedInOver13 || signedOutAge;
+    // If user is signed out, query session key to determine age.
+    // Return false (no filter), if user is over 13.
+    if (userType === 'unknown') {
+      return !signedOutOver13();
+    }
+
+    // User is signed in (student or teacher) and the filter is off.
+    // Return true (filterOn) if the user is under 13. Teachers assumed over13.
```

Comment:
super nit: I'd replace "filterOn" with something like "filter should be turned on" since a function doesn't need to know about specific variables outside of its scope.